exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentManager.java

```diff
@@ -38,36 +37,43 @@
    * @return True if the fragment has enough incoming data to be able to be run.
    * @throws FragmentSetupException, IOException
    */
-  public abstract boolean handle(RawFragmentBatch batch) throws FragmentSetupException, IOException;
+  boolean handle(RawFragmentBatch batch) throws FragmentSetupException, IOException;
 
   /**
    * Get the fragment runner for this incoming fragment. Note, this can only be requested once.
    *
    * @return
    */
-  public abstract FragmentExecutor getRunnable();
+  FragmentExecutor getRunnable();
+
+  void cancel();
 
-  public abstract void cancel();
+  /**
+   * Find out if the FragmentManager has been cancelled.
+   *
+   * @return true if the FragmentManager has been cancelled.
+   */
+  boolean isCancelled();
```

Comment:
My bad.. I did not see that mentioned in the commit message, and I did not realize the PR was made with comments in addition to the commit message.